File: src/main/java/team4/footwithme/chat/domain/ReservationChatroom.java
package team4.footwithme.chat.domain;

import jakarta.persistence.Entity;

@Entity
public class ReservationChatroom extends Chatroom {

    private Long reservationId;

    private ReservationChatroom(String name, Long reservationId) {
        super(name);
        this.reservationId = reservationId;
    }

    protected ReservationChatroom() {
    }

    public static ReservationChatroom create(String name, Long reservationId) {
        return ReservationChatroom.builder()
            .name(name)
            .reservationId(reservationId)
            .build();
    }

    public static ReservationChatroomBuilder builder() {
        return new ReservationChatroomBuilder();
    }

    public Long getReservationId() {
        return this.reservationId;
    }

    public static class ReservationChatroomBuilder {
        private String name;
        private Long reservationId;

        ReservationChatroomBuilder() {
        }

        public ReservationChatroomBuilder name(String name) {
            this.name = name;
            return this;
        }

        public ReservationChatroomBuilder reservationId(Long reservationId) {
            this.reservationId = reservationId;
            return this;
        }

        public ReservationChatroom build() {
            return new ReservationChatroom(this.name, this.reservationId);
        }

        public String toString() {
            return "ReservationChatroom.ReservationChatroomBuilder(name=" + this.name + ", reservationId=" + this.reservationId + ")";
        }
    }
}

File: src/main/java/team4/footwithme/chat/domain/ReservationChatroom.kt
package team4.footwithme.chat.domain

import jakarta.persistence.Entity

@Entity
class ReservationChatroom : Chatroom {
    var reservationId: Long? = null
        

    private constructor(name: String?, reservationId: Long?) : super(name) {
        this.reservationId = reservationId
    }

    protected constructor()

    class ReservationChatroomBuilder internal constructor() {
        private var name: String? = null
        private var reservationId: Long? = null

        fun name(name: String?): ReservationChatroomBuilder {
            this.name = name
            return this
        }

        fun reservationId(reservationId: Long?): ReservationChatroomBuilder {
            this.reservationId = reservationId
            return this
        }

        fun build(): ReservationChatroom {
            return ReservationChatroom(this.name, this.reservationId)
        }

        override fun toString(): String {
            return "ReservationChatroom.ReservationChatroomBuilder(name=" + this.name + ", reservationId=" + this.reservationId + ")"
        }
    }

    companion object {
        fun create(name: String?, reservationId: Long?): ReservationChatroom {
            return builder()
                .name(name)
                .reservationId(reservationId)
                .build()
        }

        fun builder(): ReservationChatroomBuilder {
            return ReservationChatroomBuilder()
        }
    }
}
